cmd/catchpointdump/file.go

```diff
@@ -181,6 +197,167 @@ func isGzipCompressed(catchpointReader *bufio.Reader, catchpointFileSize int64)
+	stat, err := os.Stat(catchpointFile)
+	if err != nil {
+		return fmt.Errorf("unable to stat '%s': %v", catchpointFile, err)
+	}
+	if stat.Size() == 0 {
+		return fmt.Errorf("file '%s' is empty", catchpointFile)
+	}
+
+	f, err := os.Open(catchpointFile)
+	if err != nil {
+		return fmt.Errorf("unable to open file '%s': %v", catchpointFile, err)
+	}
+	defer f.Close()
+
+	outFile := os.Stdout
+	if outFileName != "" {
+		outFile, err = os.OpenFile(outFileName, os.O_RDWR|os.O_TRUNC|os.O_CREATE, 0755)
+		if err != nil {
+			return fmt.Errorf("unable to create file '%s': %v", outFileName, err)
+		}
+		defer outFile.Close()
+	}
+
+	return rawDumpCatchpointStream(f, stat.Size(), outFile)
+}
+
+func rawDumpCatchpointStream(r io.Reader, fileSize int64, outFile *os.File) error {
+	bufRd := bufio.NewReader(r)
+	tarReader, _, err := getCatchpointTarReader(bufRd, fileSize)
+	if err != nil {
+		return err
+	}
+	var fileHeader ledger.CatchpointFileHeader
+	var fileHeaderFound bool
+	var version uint64
+
+	for {
+		hdr, err := tarReader.Next()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		fname := hdr.Name
+
+		chunkData := make([]byte, hdr.Size)
+		_, readErr := io.ReadFull(tarReader, chunkData)
+		if readErr != nil && readErr != io.EOF {
+			return readErr
+		}
+
+		switch fname {
+		case ledger.CatchpointContentFileName:
+			err = protocol.Decode(chunkData, &fileHeader)
+			if err != nil {
+				return err
+			}
+			fileHeaderFound = true
+			version = fileHeader.Version
+			printHeaderFields(fileHeader)
+
+		case "stateProofVerificationContext.msgpack":
+			// skip decoding state proof verification context
+
+		default:
+			// it might be balances.*.msgpack or something else
+			if strings.HasPrefix(fname, "balances.") && strings.HasSuffix(fname, ".msgpack") {
+				if !fileHeaderFound {
+					fmt.Fprintf(outFile, "Warning: found a balances chunk %s before content.json\n", fname)
+				}
+				if version == ledger.CatchpointFileVersionV5 {
+					var chunk ledger.CatchpointSnapshotChunkV5
+					err = protocol.Decode(chunkData, &chunk)
+					if err != nil {
+						fmt.Fprintf(outFile, "Error decoding chunk %s: %v\n", fname, err)
+						return err
+					}
+					for _, brec := range chunk.Balances {
+						// decode accountData
+						var ad basics.AccountData
+						err = protocol.Decode(brec.AccountData, &ad)
+						if err != nil {
+							fmt.Fprintf(outFile, "Error decoding account data %s: %v\n", brec.Address.String(), err)
+							return err
+						}
+						adJSON, _ := json.Marshal(ad)
+						fmt.Fprintf(outFile, "%s : %s\n", brec.Address.String(), string(adJSON))
+						// v5 has no resources map
+					}
+				} else {
+					var chunk ledger.CatchpointSnapshotChunkV6
+					err = protocol.Decode(chunkData, &chunk)
+					if err != nil {
+						fmt.Fprintf(outFile, "Error decoding chunk %s: %v\n", fname, err)
+						break
+					}
+					// Balances - only print if not onlineOnly
+					if !onlineOnly {
```

Comment:
pass an options argument to get rid of global var dependency?

Reply:
I understand, my point is this function appears to have only this `onlineOnly` global var dependency